src/imars3d/backend/preparation/normalization.py

```diff
@@ -1,48 +1,77 @@
-    darks:
-        3D array of dark field images, axis=0 is the image number axis.
-    cut_off:
-        Permitted maximum value for the normalized data, originated from tomopy, negative values means no cutoff.
-    ncore:
-        number of cores to use for parallel processing, default is -1, which means using all available cores.
-
-    Returns
-    -------
-        normalized 3D array of images, the first dimension is the rotation angle omega.
-    """
-    # parse input (mostly for Tomopy)
-    cut_off = None if cut_off < 0 else float(cut_off)
-    ncore = None if ncore == -1 else int(ncore)
-    # use median filter to remove outliers from flats and darks
-    # NOTE: this will remove the random noises coming from the environment.
-    flats = np.median(flats, axis=0)
-    darks = np.median(darks, axis=0)
-    # apply normalization
-    # NOTE:
-    # For pixels where dark > flat, tomopy replace the value with 1e-6 on-the-fly
-    # see https://github.com/tomopy/tomopy/blob/master/source/tomopy/prep/normalize.py#L135
-    #
-    arrays_normalized = tomopy.normalize(arrays, flats, darks, cutoff=cut_off, ncore=ncore)
-    # return
-    return arrays_normalized
+logger = param.get_logger(__name__)
+logger.name = __name__
+
+
+class normalization(param.ParameterizedFunction):
+    arrays = param.Array(doc="3D array of images, the first dimension is the rotation angle omega.")
+    flats = param.Array(
+        doc="3D array of flat field images (aka flat field, open beam), axis=0 is the image number axis."
+    )
+    darks = param.Array(doc="3D array of dark field images, axis=0 is the image number axis.")
+    cut_off = param.Number(
+        default=-1.0,
+        doc="Permitted maximum value for the normalized data, originated from tomopy, negative values means no cutoff.",
+    )
+    max_workers = param.Integer(
+        default=0,
+        bounds=(0, max(1, multiprocessing.cpu_count() - 2)),
+        doc="Maximum number of processes allowed during loading",
+    )
+
+    def __call__(self, **params):
+        """
+        Normalize the input array(s) by subtracting the dark field and dividing by the adjusted flat field.
+
+        Parameters
+        ----------
+        arrays:
+            3D array of images, the first dimension is the rotation angle omega.
+        flats:
+            3D array of flat field images (aka flat field, open beam), axis=0 is the image number axis.
+        darks:
+            3D array of dark field images, axis=0 is the image number axis.
+        cut_off:
+            Permitted maximum value for the normalized data, originated from tomopy, negative values means no cutoff.
+        max_workers:
+            number of cores to use for parallel processing, default is -1, which means using all available cores.
+
+        Returns
+        -------
+            normalized 3D array of images, the first dimension is the rotation angle omega.
+        """
```

Comment:
Could you move this docstring to the class level?  Sphinx will not render this docstring for normalization if it is under __call__.